### PR TITLE
Allow templated expansion of data dependency names in the RPM spec file.

### LINKED ISSUE #4810: Fedora RPM fails to build
### Description of the problem / feature request:

Building //scripts/packages/fedora:bazel fails

### Feature requests: what underlying problem are you trying to solve with this feature?

Attempting to build the Fedora RPM.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

    bazel build //scripts/packages/fedora:bazel

### What operating system are you running Bazel on?

Fedora 27

### What's the output of `bazel info release`?

    release 0.11.1- (@non-git)

### If `bazel info release` returns "development version" or "(@non-git)", tell us how you built Bazel.

Installed from copr/vbatts yum repo:
https://copr-be.cloud.fedoraproject.org/results/vbatts/bazel/fedora-$releasever-$basearch/

### What's the output of `git remote get-url origin ; git rev-parse master ; git rev-parse HEAD` ?

git@github.com:bazelbuild/bazel.git
a2fc4e3488ea4baff083a35a2351d972b8cd12da
a2fc4e3488ea4baff083a35a2351d972b8cd12da

###  Have you found anything relevant by searching the web?

- Github explains how to build the package: https://github.com/bazelbuild/bazel/issues/2701

### Any other information, logs, or outputs that you want to share?

```
$ bazel build //scripts/packages/fedora:bazel
INFO: Analysed target //scripts/packages/fedora:bazel (0 packages loaded).
INFO: Found 1 target...
INFO: From MakeRpm scripts/packages/fedora/bazel-x86_64.rpm:
Building RPM for bazel at bazel-out/k8-fastbuild/bin/scripts/packages/fedora/bazel-x86_64.rpm
Error calling rpmbuild:
Executing(%prep): /bin/sh -e /tmp/tmpCxC7Ej/TMP/rpm-tmp.UniZrT
+ umask 022
+ cd /tmp/tmpCxC7Ej/BUILD
+ exit 0
Executing(%build): /bin/sh -e /tmp/tmpCxC7Ej/TMP/rpm-tmp.bH2r8f
+ umask 022
+ cd /tmp/tmpCxC7Ej/BUILD
+ exit 0
Executing(%install): /bin/sh -e /tmp/tmpCxC7Ej/TMP/rpm-tmp.ECZqPC
+ umask 022
+ cd /tmp/tmpCxC7Ej/BUILD
+ '[' /tmp/tmpCxC7Ej/BUILDROOT/bazel-devel-1.x86_64 '!=' / ']'
+ rm -rf /tmp/tmpCxC7Ej/BUILDROOT/bazel-devel-1.x86_64
++ dirname /tmp/tmpCxC7Ej/BUILDROOT/bazel-devel-1.x86_64
+ mkdir -p /tmp/tmpCxC7Ej/BUILDROOT
+ mkdir /tmp/tmpCxC7Ej/BUILDROOT/bazel-devel-1.x86_64
+ export DONT_STRIP=1
+ DONT_STRIP=1
+ mkdir -p /tmp/tmpCxC7Ej/BUILDROOT/bazel-devel-1.x86_64/usr/bin/
+ install -m 755 bazel /tmp/tmpCxC7Ej/BUILDROOT/bazel-devel-1.x86_64/usr/bin/bazel
install: cannot stat 'bazel': No such file or directory
error: Bad exit status from /tmp/tmpCxC7Ej/TMP/rpm-tmp.ECZqPC (%install)


RPM build errors:
    Bad exit status from /tmp/tmpCxC7Ej/TMP/rpm-tmp.ECZqPC (%install)

No RPM file created.
ERROR: /home/nelson/src/ext/bazel/scripts/packages/fedora/BUILD:10:1: output 'scripts/packages/fedora/bazel-x86_64.rpm' was not created
ERROR: /home/nelson/src/ext/bazel/scripts/packages/fedora/BUILD:10:1: not all outputs were created or valid
Target //scripts/packages/fedora:bazel failed to build
Use --verbose_failures to see the command lines of failed build steps.
INFO: Elapsed time: 0.720s, Critical Path: 0.24s
FAILED: Build did NOT complete successfully
```

### PR DESCRIPTION
**Pull request by katre**
Fixes #4810, invalidates #4782, and extends #4747.

Change-Id: I19426bb30a151358875e9bae0e0c1d92b3818432

### CODE REVIEW

**Comment by katre** [comment]
@mmikitka, can you also take a look?

**Comment by laszlocsomor** [comment]
Sorry I can't look at it today, only on Monday.

File: tools/build_defs/pkg/rpm.bzl [hunk 1]
```diff
@@ -101,6 +134,8 @@ Args:
   spec_file: The RPM spec file to use. If the version or version_file
     attributes are provided, the Version in the spec will be overwritten.
     Any Sources listed in the spec file must be provided as data dependencies.
+    The base names of data dependencies can be replaced with the actual location
+   using "{basename}" syntax.
```
Comment (laszlocsomor) [inline tools/build_defs/pkg/rpm.bzl]: nit: match indentation of line above

Comment (katre) [reply tools/build_defs/pkg/rpm.bzl]: Done.

**Pull request closed without merging**